app/services/calculate_status.rb

```diff
@@ -1,39 +1,49 @@
 class CalculateStatus
   UPDATE_TIME = 1
 
+  attr_reader :silent
+  def initialize(silent: Rails.env.test?)
+    @silent = silent
+  end
+
+  def log(msg)
+    puts "[#{Time.now}] #{msg}" unless silent
+  end
+
   def call
-    puts "[#{Time.now}] Going to calculate the statuses"
     duration = Benchmark.ms do
-      puts "[#{Time.now}] Calculating the Crowdsource status"
+      log 'Calculating the Crowdsource status'
       StatusCrowdsource
         .joins(store: :status_crowdsource_users)
         .where('status_crowdsource_users.created_at > ?', 2.hours.ago)
         .find_each.with_index do |s, i|
-        puts "Calculated #{i}" if (i % 100).zero?
+        log "Calculated #{i}" if (i % 100).zero?
```

Comment:
Shouldn't we use rails logger? Is the output of `puts` going to the logfiles?

Reply:
I'm not sure. Since this is mostly used in a rake task, I think these puts serve mostly as something for you to see when you run it, to make sure it's not stuck.